junit-jupiter-api/src/main/java/org/junit/jupiter/api/extension/ExtensionContext.java

```diff
@@ -744,6 +768,11 @@ public Namespace append(Object... parts) {
 			Collections.addAll(newParts, parts);
 			return new Namespace(newParts);
 		}
+
+		@API(status = INTERNAL, since = "5.13")
+		public Object[] getParts() {
+			return parts.toArray();
```

Comment:
When we switch to JDK 17 in 6.0, returning a `List` here will allow us to avoid copying since we can use `List.copyOf()`. Thus, I think we should do so right away and introduce a `org.junit.platform.engine.support.store.Namespace.create(List)` method.

```suggestion
		public List<Object> getParts() {
			return unmodifiableList(parts);
```

Reply:
Done in https://github.com/junit-team/junit5/pull/4281/commits/78a6023867f039fa78ef75391d5953dda4cdb660 !
Would it be okay to ask if this aligns with your intention? 🧐